src/treq/testing.py

```diff
@@ -239,7 +239,17 @@ def __init__(self, resource):
                 function = _reject_files(function)
 
             setattr(self, function_name, function)
-        self.flush = _agent.flush
+        self.flush = self._agent.flush
+
+    def cleanSessions(self):
+        """
+        Clean up sessions to prevent leaving behind a dirty reactor.
```

Comment:
The behavior of leaving a dirty reactor looks like a bug in Twisted to me, actually. The session's delayed call should be registered with the reactor passed to `Site`.

Reply:
I filed https://twistedmatrix.com/trac/ticket/10177 for this issue.

I think that it could be worked around here by overriding the site's [`sessionFactory`](https://github.com/twisted/twisted/blob/5d0badef0fce092af325c51063f43072af3d4dff/src/twisted/web/server.py#L773) to pass the memory reactor to the `Session` constructor. Then you wouldn't need this method.